R/utilities.R

```diff
@@ -4101,3 +4104,74 @@ supplement_maf <- function(incoming_maf,
                    missing_sample_maf)
   return(full_maf)
 }
+
+#' INTERNAL FUNCTION called by collate_results, not meant for out-of-package usage.
+#' Identify samples with mutated IGHV from MiXCR + IgBLAST results
+#'
+#' @param sample_table        A data frame with sample_id as the first column.
+#' @param results_directory   Directory containging MiXCR/IgBLASTN results. Depends on files having the columns "mutatedStatus" and "numMissingRegions"
+#' @param missing_threshold   Limit for missing IGH regions during MiXCR assembly. Default is 2. Higher thresholds may return less accurate mutation status.
+#'
+#' @return Samples table.
+#' @import tidyverse
+#'
+#' @examples 
+#' sample_table = collate_ighv_results(sample_table = sample_table, results_directory=".../01-mixcr/{seq_type}/", missing_threshold=3)
+#'
+
+collate_igblast_results = function(sample_table,
+                                seq_type_filter = "genome",
+                                results_directory = "",
+                                missing_threshold = 2){
+  if (missing(sample_table)) {
+    sample_table = get_gambl_metadata(seq_type_filter = seq_type_filter) %>%
+                    dplyr::select(sample_id, patient_id, biopsy_id)
+  }
+  
+  # Check seq_type_filter is genome or mrna
+  if (seq_type_filter=="capture") {
+    message("'capture' seq_type_filter doesn't make sense for this tool. Use 'genome' or 'mrna'.")
+    return (sample_table)
+  }
+  # Look in mixcr-1.2/results if no results directory specified
+  if (results_directory == "") {
+    base = config::get("project_base")
+    results_directory = glue(paste0(base, "gambl/mixcr-1.2/01-mixcr/{seq_type_filter}/"))
+  }
+  print(paste0("Searching for MiXCR results in directory:", results_directory))
+
+  # Make table with necessary columns from result files
+  mixcr_table = data.frame(biopsy_id=character(),
+                            mixcr_mutated=character(),
+                            missing=double())
```

Comment:
Start with a directory listing to avoid a loop. You probably will have to do this twice (once for each unix group). I think another recently added collate function does something similar